roles/prometheus/templates/prometheus.yml.j2

```diff
@@ -17,8 +17,10 @@ remote_read:
   {{ prometheus_remote_read | to_nice_yaml(indent=2,sort_keys=False) | indent(2, False) }}
 {% endif %}
 
+{% if prometheus_rules_enable and not prometheus_agent_mode %}
```

Comment:
`rule_files` are not allowed in agent mode
```Aug 17 20:25:51 master2-int prometheus[50413]: ts=2023-08-17T20:25:51.526Z caller=main.go:479 level=error msg="Error loading config (--config.file=/etc/prometheus/prometheus.yml)" file=/etc/prometheus/prometheus.yml err="field rule_files is not allowed in agent mode"```

Reply:
You probably want to skip copying the rules when using agent mode and not just configure the service to ignore the rules.

https://github.com/prometheus-community/ansible/blob/93ed29132cd8ff042f7ba612eb8cd1eb6e20991f/roles/prometheus/tasks/configure.yml#L1-L25

Also I feel that the addition of the `prometheus_rules_enable` variable is unnecessary.
I'm unsure if we need to be able to disable the rules in setups other than agent mode.
But if we want to offer that option a better approach would be to just leave the `prometheus_alert_rules` and `prometheus_alert_rules_files` variables empty/unconfigured and disable the rules based on that.